src/transformers/trainer.py

```diff
@@ -2388,7 +2394,11 @@ def _save_checkpoint(self, model, trial, metrics=None):
                     torch.save(self.scaler.state_dict(), os.path.join(output_dir, SCALER_NAME))
         elif self.args.should_save and not self.deepspeed:
             # deepspeed.save_checkpoint above saves model/optim/sched
-            torch.save(self.optimizer.state_dict(), os.path.join(output_dir, OPTIMIZER_NAME))
+            if self.fsdp:
+                torch.save(full_osd, os.path.join(output_dir, OPTIMIZER_NAME))
```

Comment:
saving on rank 0 should be efficient and enough, right?

Reply:
I believe `self.args.should_save` in this case is already handling saving on rank 0

Reply:
oh, okay, got it, thank you!